pkg/render/whisker/component.go

```diff
@@ -0,0 +1,427 @@
+			Name:      WhiskerDeploymentName,
+			Namespace: WhiskerNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.ToPtr(int32(1)),
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: WhiskerDeploymentName,
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:       c.cfg.Installation.ControlPlaneNodeSelector,
+					ServiceAccountName: WhiskerServiceAccountName,
+					Tolerations:        tolerations,
+					ImagePullSecrets:   secret.GetReferenceList(c.cfg.PullSecrets),
+					Containers:         ctrs,
+					Volumes:            volumes,
+				},
+			},
+		},
+	}
+}
+
+func secretMount(path string, scrt *corev1.Secret) corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      fmt.Sprintf("%s-%s", scrt.Name, "scrt"),
+		MountPath: path,
+	}
+}
+
+func secretVolume(scrt *corev1.Secret) corev1.Volume {
+	return corev1.Volume{
+		Name:         fmt.Sprintf("%s-%s", scrt.Name, "scrt"),
+		VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: scrt.Name}},
+	}
+}
+
+func (c *Component) guardianService() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GuardianServiceName,
+			Namespace: WhiskerNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"k8s-app": WhiskerDeploymentName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: "linseed",
+					Port: 443,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+				{
+					Name: "elasticsearch",
+					Port: 9200,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+				{
+					Name: "kibana",
+					Port: 5601,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+		},
+	}
+}
+
+func (c *Component) clusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: WhiskerClusterRoleName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     WhiskerClusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      WhiskerClusterRoleName,
+				Namespace: WhiskerNamespace,
+			},
+		},
+	}
+}
+
+func (c *Component) clusterRole() *rbacv1.ClusterRole {
+	policyRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"*"},
+			Resources: []string{"*"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{"felixconfigurations"},
```

Comment:
Hm, what accesses felixconfigurations from this deployment?

Reply:
UI APIs seems to need it when it's connected.

Reply:
Removed!